client/me/purchases/payment-methods/main.tsx

```diff
@@ -18,6 +19,7 @@ function PaymentMethods() {
 		<Main wideLayout className="payment-methods__main">
 			<DocumentHead title={ titles.paymentMethods } />
 			<PageViewTracker path="/me/purchases/payment-methods" title="Me > Payment Methods" />
+			<QueryUserPurchases />
```

Comment:
Note to future readers (and myself): we really need to replace this query component with a react-query hook. Query components do not provide much control and there's no way to tell why one exists (which is why it took such a long time to [remove outdated ones in checkout](https://github.com/Automattic/wp-calypso/pull/87540)), leading to decreased performance and subtle bugs if the query component is not rendered when the child that needs it is (which often happens as components get reorganized and reused over the years).

Reply:
Oh wait, it looks like those hooks don't actually return anything... they just do the same thing as the query component. Hm... I guess that's slightly better than the query component but not by a lot. Well, I leave it to you to decide if you want to implement that change or not. I'm probably going to go through and change these to not use Redux anymore soon anyway. 🤷

Reply:
Oh, interesting - thanks for taking a look! I think I'll leave it for now then if you're going to change them soon. I'm not _toooo_ familiar with TypeScript, but I believe that I'd need to change [this](https://github.com/Automattic/wp-calypso/blob/8f8cfb8e870f44f5def90d227a5a75759cb95c1a/client/me/purchases/payment-methods/payment-method-list.tsx) into a functional component to use them properly, which is probably a little out-of-scope.

Reply:
To close the loop, I'm starting this process in https://github.com/Automattic/wp-calypso/pull/96367